src/StringManipulation/BackspaceCompareOnSpace.java

```diff
@@ -0,0 +1,86 @@
+package StringManipulation;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+Given two strings S and T, return if they are equal when both are typed into empty text editors. # means a backspace character.
+
+Example 1:
+
+Input: S = "ab#c", T = "ad#c"
+Output: true
+Explanation: Both S and T become "ac".
+Example 2:
+
+Input: S = "ab##", T = "c#d#"
+Output: true
+Explanation: Both S and T become "".
+Example 3:
+
+Input: S = "a##c", T = "#a#c"
+Output: true
+Explanation: Both S and T become "c".
+Example 4:
+
+Input: S = "a#c", T = "b"
+Output: false
+Explanation: S becomes "c" while T becomes "b".
+Note:
+
+1 <= S.length <= 200
+1 <= T.length <= 200
+S and T only contain lowercase letters and '#' characters.
+*/
+public class BackspaceCompareOnSpace {
+	public static boolean backspaceCompare(String S, String T) {
+	 List<Character> s = new ArrayList<Character>();
```

Comment:
I used stringbuilder. need to find out if using list vs stringbuilder has any advantage.

Reply:
They are similar if I am not wrong, StringBuilder/ Buffer creates char arrays underneath the hood and finally mutate into a String object

Reply:
good to know.